src/main/java/gift/model/product/Option.java

```diff
@@ -0,0 +1,112 @@
+package gift.model.product;
+
+import gift.model.BaseTimeEntity;
+import jakarta.persistence.Column;
+import jakarta.persistence.Entity;
+import jakarta.persistence.FetchType;
+import jakarta.persistence.GeneratedValue;
+import jakarta.persistence.GenerationType;
+import jakarta.persistence.Id;
+import jakarta.persistence.JoinColumn;
+import jakarta.persistence.ManyToOne;
+import jakarta.validation.constraints.NotNull;
+import java.util.regex.Pattern;
+
+@Entity
+public class Option extends BaseTimeEntity {
+
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    private Long id;
+
+    @NotNull
+    private String name;
+
+    @NotNull
+    @Column(nullable = false)
+    private Integer quantity;
+
+    @ManyToOne(fetch = FetchType.LAZY)
+    @JoinColumn(name = "product_id")
+    private Product product;
+
+    protected Option() {
+    }
+
+    public Option(String name, Integer quantity, Product product) {
+        OptionNameValidator.isValidName(name);
+        OptionNameValidator.isValidCount(quantity);
+        this.name = name;
+        this.quantity = quantity;
+        this.product = product;
+    }
+
+    public Long getId() {
+        return id;
+    }
+
+    public String getName() {
+        return name;
+    }
+
+    public Integer getQuantity() {
+        return quantity;
+    }
+
+    public void update(String name, Integer quantity) {
+        OptionNameValidator.isValidName(name);
+        OptionNameValidator.isValidCount(quantity);
+        this.name = name;
+        this.quantity = quantity;
+    }
+
+    public boolean isSameName(String name) {
+        return this.name.equals(name);
+    }
+
+    //-- bussiness logic --//
+    public void purchase(int count) {
+        if (quantity < count) {
+            throw new IllegalArgumentException("재고가 부족합니다.");
+        }
+        quantity -= count;
+    }
+
+
+    //-- validation --//
+    private static class OptionNameValidator {
+
+        private static final int MAX_LENGTH = 50;
+        private static final Pattern VALID_PATTERN = Pattern.compile(
+            "^[a-zA-Z0-9가-힣\\(\\)\\[\\]\\+\\-\\&\\/\\_\\s]*$");
+
+        public static boolean isValidName(String value) {
+            if (value.isBlank()) {
+                throw new IllegalArgumentException("옵션 이름은 필수 입력값입니다.");
+            }
+
+            if (value.length() > MAX_LENGTH) {
+                throw new IllegalArgumentException("옵션 이름은 최대 50자까지 입력 가능합니다.");
+            }
+
+            if (!VALID_PATTERN.matcher(value).matches()) {
+                throw new IllegalArgumentException(
+                    "옵션 이름은 영문, 한글, 숫자, 공백,  ( ), [ ], +, -, &, /, _만 입력 가능합니다.");
+            }
+
+            return true;
+        }
+
+        public static boolean isValidCount(Integer value) {
+            if (value < 0) {
```

Comment:
요렇게 바뀌어야 하지 않을까요ㅎㅎ 테스트의 중요성, 테스트 중에서도 경계값 테스트의 중요성이 보이는 부분이네요!
```suggestion
            if (value <= 0) {
```

Reply:
감사합니다 확실히 테스트의 중요성을 다시한번 깨닫네요..